### PR TITLE
Adding swagger doc

### PR DESCRIPTION
**Pull request by tflucker**
# Summary
created rough draft of swagger doc to show design of APIs and to provide a technical project artifact.

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+      responses:
+        "200": 
+          description: "Successful reponse"
+          schema:
+            $ref: "#/definitions/MessageResponse"
+        "400":
+          description: "Invalid Response, username already exists."
+          schema:
+            $ref: "#/definitions/MessageResponse"
+  /bankAccount:
```
Comment (jdorancy-bu) [inline swagger.yaml:60]: Can we name paths with hyphens instead of camel case?
Example: `bank-account` instead of `bankAccount`

**Review by jdorancy-bu — COMMENTED** [review]
- I did an initial review and looks good so far.
- Now, when it comes to requirements and communication with stakeholders I'm thinking we will need to create API contract documents which is going to spell out all the field, description and constraint for each API endpoint which the stakeholder will need to sign off on. We will use those contracts to guide the changes that we need to make here.

File: swagger.yaml [hunk 2]
```diff
@@ -0,0 +1,445 @@
...
+  version: "1.0.0"
+  title: "Trackr - Personal Transaction Management"
+  termsOfService: "http://swagger.io/terms/"
+  contact:
+    email: "tflucker@bu.edu"
+  license:
+    name: "Apache 2.0"
+    url: "http://www.apache.org/licenses/LICENSE-2.0.html"
+host: "trackr.swagger.io"
+basePath: "/v2"
```
[the remark this answers is not in the document]
Comment (tflucker) [reply swagger.yaml:13]: This was part of the default configuration when you create a new document. I can remove it if you would like.

[the remark this answers is not in the document]
Comment (tflucker) [reply swagger.yaml:13]: whoops, i misread your comment. I'll modify it to be "/v1"

**Review by jdorancy-bu — APPROVED** [review]
🚀

**Pull request merged**